src/wallet/rpcdump.cpp

```diff
@@ -804,6 +804,91 @@ UniValue dumpwallet(const JSONRPCRequest& request)
     return reply;
 }
 
+struct ImportData
+{
+    // Input data
+    std::unique_ptr<CScript> redeemscript; //! Provided redeemScript; will be mvoed to `import_scripts` if relevant.
+    std::unique_ptr<CScript> witnessscript; //! Provided witnessScript; will be moved to `import_scripts` if relevant.
+    std::map<CKeyID, CPubKey> pubkeys;
+    std::map<CKeyID, CKey> privkeys;
+
+    // Output data
+    std::set<CScript> import_scripts;
```

Comment:
Instead of a `std::set`, could `std::vector` be used for these in order to preserve the order in which things will be imported? This is useful for #14075 where we want to have the order in which things are added to the wallet be the order that was specified in the import.

Reply:
A few more changes would be required, see a fairly minimal change set(I made vectors of all the sets): https://gist.github.com/instagibbs/bd11e8666d929d84e4abc2214962e752

Reply:
This has been implemented in #14075 as that is where this behavior is desired.

Reply:
I think it's always desired, particularly when someone imports private keys, as those would get added to the keypool in order of their occurrence in the wallet IIUC.

Reply:
@Sjors As of this PR, there is no reason for doing so as we can't import into the keypool yet. Follow-ups can add the ordering.